index.js

```diff
@@ -84,26 +85,35 @@ cli.on('respawn', function(flags, child) {
 });
 
 function run() {
-  cli.launch({
+  var envOpts = {
     cwd: opts.cwd,
     configPath: opts.gulpfile,
     require: opts.require,
     completion: opts.completion,
-  }, handleArguments);
+  };
+
+  cli.launch(envOpts, function(env) {
+    var config;
+    try {
+      config = loadConfigFiles(env.configFiles['.gulp'], ['home', 'cwd']);
```

Comment:
Do we need to validate the config?  Could we just ignore and log a warning or error if it's the wrong type?

I don't like the throw/try/catch pattern and we shouldn't be exiting.